File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/Dog.java
package io.quarkus.it.panache;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

// custom id type
@Entity
public class Dog extends PanacheEntityBase {

    @Id
    @GeneratedValue
    public Integer id;

    public String name;

    public String race;

    @ManyToOne
    public Person owner;

    public Dog(String name, String race) {
        this.name = name;
        this.race = race;
    }

    public Dog() {
    }

}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/Dog.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheCompanionBase
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntityBase
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne

@Entity
class Dog() : PanacheEntityBase {
    constructor(name: String, race: String) : this() {
        this.name = name
        this.race = race
    }

    @Id
    @GeneratedValue
    var id: Int? = null

    lateinit var name: String

    lateinit var race: String

    @ManyToOne
    lateinit var owner: Person

    companion object : PanacheCompanionBase<Dog, Int>
}
